dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/Common/GeminiChatCompletionClient.cs

```diff
@@ -56,7 +57,7 @@ internal class GeminiChatCompletionClient : GeminiClient, IGeminiChatCompletionC
         PromptExecutionSettings? executionSettings = null,
         CancellationToken cancellationToken = default)
     {
-        ValidateAndPrepareChatHistory(chatHistory);
+        ValidateAndPrepareChatHistory(ref chatHistory);
```

Comment:
I don't think `ref` is necessary as `ChatHistory` is already a class.

Reply:
It is necessary because I reassigned variable chat history in ValidateAndPrepareChatHistory method with copy of chathistory. This is deliberate in order to manipulate the chathistory object without altering the passed object.